src/mkpsxiso/iso.cpp

```diff
@@ -354,18 +358,22 @@ int iso::DirTreeClass::CalculateTreeLBA(int lba)
 				entry.subdir->name = entry.id;
 			}
 
-			lba += GetSizeInSectors(entry.subdir->CalculateDirEntryLen());
+			lba += GetSizeInSectors(entry.subdir->CalculateDirEntryLen());		
 		}
 		else
 		{
 			// Increment LBA by the size of file
 			if ( entry.type == EntryType::EntryFile || entry.type == EntryType::EntryXA_DO || entry.type == EntryType::EntryDummy )
-			{
-				lba += GetSizeInSectors(entry.length, 2048);
+			{	
+				lba += (entry.flba)
+					? entry.flba - lba + GetSizeInSectors(entry.length, 2048)
+					: GetSizeInSectors(entry.length, 2048);					
 			}
 			else if ( entry.type == EntryType::EntryXA )
 			{
-				lba += GetSizeInSectors(entry.length, 2336);
+				lba += (entry.flba)
+					? entry.flba - lba + GetSizeInSectors(entry.length, 2336)
+					: GetSizeInSectors(entry.length, 2336);					

```

Comment:
yes, esentially this is lbe = entry.flba + size if it exists, I wasn't sure what would happen if lba wasn't properly tracked (even here I'm not sure) so instead of just adding size to the end I'm also moving it to the last flba, tho in retrospect, this might still be a problem if the last files isn't the one with the highest offset

Reply:
Added a safer way to track final lba with offsets